echo_handlers.go

```diff
@@ -239,97 +242,28 @@ func getEndOfBudgetString(endOfBudget time.Time) (result string) {
-		appName = c.QueryParam("c")
-	}
-	pubkey := c.QueryParam("pubkey")
-	returnTo := c.QueryParam("return_to")
-	maxAmount := c.QueryParam("max_amount")
-	budgetRenewal := strings.ToLower(c.QueryParam("budget_renewal"))
-	expiresAt := c.QueryParam("expires_at") // YYYY-MM-DD or MM/DD/YYYY or timestamp in seconds
-	if expiresAtTimestamp, err := strconv.Atoi(expiresAt); err == nil {
-		expiresAt = time.Unix(int64(expiresAtTimestamp), 0).Format(time.RFC3339)
-	}
-	expiresAtISO, _ := time.Parse(time.RFC3339, expiresAt)
-	expiresAtFormatted := expiresAtISO.Format("January 2, 2006 03:04 PM")
-
-	requestMethods := c.QueryParam("request_methods")
-	customRequestMethods := requestMethods
-	if requestMethods == "" {
-		// if no request methods are given, enable them all by default
-		keys := []string{}
-		for key := range nip47MethodDescriptions {
-			keys = append(keys, key)
-		}
-
-		requestMethods = strings.Join(keys, " ")
-	}
+func (svc *Service) CSRFHandler(c echo.Context) error {
 	csrf, _ := c.Get(middleware.DefaultCSRFConfig.ContextKey).(string)
-
-	user, err := svc.GetUser(c)
-	if err != nil {
-		return err
-	}
-	if user == nil {
-		sess, _ := session.Get(CookieName, c)
-		sess.Values["return_to"] = c.Path() + "?" + c.QueryString()
-		sess.Options.MaxAge = 0
-		sess.Options.SameSite = http.SameSiteLaxMode
-		if svc.cfg.CookieDomain != "" {
-			sess.Options.Domain = svc.cfg.CookieDomain
-		}
-		sess.Save(c.Request(), c.Response())
-		return c.Redirect(302, fmt.Sprintf("/%s/auth?c=%s", strings.ToLower(svc.cfg.LNBackendType), appName))
-	}
-
-	//construction to return a map with all possible permissions
-	//and indicate which ones are checked by default in the front-end
-	type RequestMethodHelper struct {
-		Description string
-		Icon        string
-		Checked     bool
-	}
-
-	requestMethodHelper := map[string]*RequestMethodHelper{}
-	for k, v := range nip47MethodDescriptions {
-		requestMethodHelper[k] = &RequestMethodHelper{
-			Description: v,
-			Icon:        nip47MethodIcons[k],
-		}
-	}
-
-	for _, m := range strings.Split(requestMethods, " ") {
-		if _, ok := nip47MethodDescriptions[m]; ok {
-			requestMethodHelper[m].Checked = true
-		}
-	}
-
-	return c.Render(http.StatusOK, "apps/new.html", map[string]interface{}{
-		"User":                 user,
-		"Name":                 appName,
-		"Pubkey":               pubkey,
-		"ReturnTo":             returnTo,
-		"MaxAmount":            maxAmount,
-		"BudgetRenewal":        budgetRenewal,
-		"ExpiresAt":            expiresAt,
-		"ExpiresAtFormatted":   expiresAtFormatted,
-		"RequestMethods":       requestMethods,
-		"CustomRequestMethods": customRequestMethods,
-		"RequestMethodHelper":  requestMethodHelper,
-		"Csrf":                 csrf,
+	return c.JSON(http.StatusOK, &CSRFResponse{
+		Csrf: csrf,
 	})
 }
 
 func (svc *Service) AppsCreateHandler(c echo.Context) error {
 	user, err := svc.GetUser(c)
 	if err != nil {
-		return err
+		return c.JSON(http.StatusBadRequest, ErrorResponse{
```

Comment:
we do these same checks on all the endpoints. Can we add a middleware for endpoints that require the user to be logged in?